android/java/org/chromium/chrome/browser/BraveActivity.java

```diff
@@ -99,9 +121,99 @@ public void onPause() {
+            String description = "Notification channel for Brave Browser";
+            int importance = NotificationManager.IMPORTANCE_DEFAULT;
+            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
+            channel.setDescription(description);
+            // Register the channel with the system; you can't change the importance
+            // or other notification behaviors after this
+            NotificationManager notificationManager = getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    private void setupBraveSetDefaultBrowserNotification() {
+        Context context = ContextUtils.getApplicationContext();
+        if (BraveSetDefaultBrowserNotificationService.isBraveSetAsDefaultBrowser(this)) {
+            // Don't ask again
+            return;
+        }
+        Intent intent = new Intent(context, BraveSetDefaultBrowserNotificationService.class);
+        context.sendBroadcast(intent);
+    }
+
     private boolean isNoRestoreState() {
         return ContextUtils.getAppSharedPreferences().getBoolean(PREF_CLOSE_TABS_ON_EXIT, false);
     }
 
+    private void handleBraveSetDefaultBrowserDialog() {
+        /* (Albert Wang): Default app settings didn't get added until API 24
+         * https://developer.android.com/reference/android/provider/Settings#ACTION_MANAGE_DEFAULT_APPS_SETTINGS
+         */
+        Intent browserIntent = new Intent("android.intent.action.VIEW", Uri.parse("http://"));
+        boolean supportsDefault = Build.VERSION.SDK_INT >= 24;
+        ResolveInfo resolveInfo = getPackageManager().resolveActivity(
+                browserIntent, supportsDefault ? PackageManager.MATCH_DEFAULT_ONLY : 0);
+        Context context = ContextUtils.getApplicationContext();
+        if (BraveSetDefaultBrowserNotificationService.isBraveSetAsDefaultBrowser(this)) {
+            Toast toast = Toast.makeText(
+                    context, R.string.brave_already_set_as_default_browser, Toast.LENGTH_LONG);
+            toast.show();
+            return;
+        }
+        if (supportsDefault) {
+            if (resolveInfo.activityInfo.packageName.equals("com.google.android.setupwizard")
+                    || resolveInfo.activityInfo.packageName.equals("android")) {
+                LayoutInflater inflater = getLayoutInflater();
+                View layout = inflater.inflate(R.layout.brave_set_default_browser_dialog,
+                        (ViewGroup) findViewById(R.id.brave_set_default_browser_toast_container));
+
+                Toast toast = new Toast(context);
+                toast.setDuration(Toast.LENGTH_LONG);
+                toast.setView(layout);
+                toast.setGravity(Gravity.TOP, 0, 40);
+                toast.show();
+                Intent intent =
+                        new Intent(Intent.ACTION_VIEW, Uri.parse("http://www.brave.com/blog"));
```

Comment:
I would also make that hardcode as final static. I know it has been probably missed in original review, but we have a better chance to make it right now :)

Reply:
yeah, totally agree! :)
Hardcode values are all addressed in https://github.com/brave/brave-core/pull/3724/commits/8f252d2ea97c505d715bc8233da10dd7b823ae81